okhttp/src/main/java/com/squareup/okhttp/Cache.java

```diff
@@ -142,12 +142,17 @@ public Cache(File directory, long maxSize) {
     cache = DiskLruCache.create(directory, VERSION, ENTRY_COUNT, maxSize);
   }
 
-  private static String urlToKey(Request request) {
+  private static String generateEntryKey(Request request) {
+    //If the request is for partial content, we want a unique key for the partial response
```

Comment:
This is not how I'd like to implement partial responses. If we request ranges 0–100 and 100–200 of `http://square.com`, that should yield one 200-byte cache entry and not two 100-byte cache entries. I don't think this change will increase cache hit rates.